lib/packwerk/application_validator.rb

```diff
@@ -65,31 +65,42 @@ def check_autoload_path_cache
     def check_package_manifests_for_privacy
       privacy_settings = package_manifests_settings_for("enforce_privacy")
 
-      autoload_paths = @configuration.load_paths
-
       resolver = ConstantResolver.new(
         root_path: @configuration.root_path,
-        load_paths: autoload_paths
+        load_paths: @configuration.load_paths
       )
 
+      package_set = Packwerk::PackageSet.load_all_from(@configuration.root_path, package_pathspec: package_glob)
+
       errors = []
 
       privacy_settings.each do |filepath, setting|
         next unless setting.is_a?(Array)
 
+        this_package = package_set.package_from_path(relative_path(filepath))
+
         setting.each do |constant|
           # make sure the constant can be loaded
           constant.constantize # rubocop:disable Sorbet/ConstantsFromStrings
 
+          # make sure the constant can be resolved
           constant_context = resolver.resolve(constant)
-          next if constant_context
 
-          explicit_filepath = (constant.start_with?("::") ? constant[2..-1] : constant).underscore + ".rb"
+          if constant_context
```

Comment:
`check_package_manifests_for_privacy` was already quite long, and now it has two more variables and an additional branch of logic.  How about we move the `if` block into its own method?  That allows us to move both `package_set` and `this_package` along with it.

The `else` block could move into a method as well.

Reply:
Good point.

Reply:
`package_set` I wouldn't want to recompute for each package, but I guess we can memoize it

Reply:
I think `package_set` is only used in the new code for detecting cross-package privatization™. If you're moving all of that code inside a new method, you can load all package sets within it.

Reply:
```ruby
def verify_private_constant_origin(filepath, constant_context)
  package_set = Packwerk::PackageSet.load_all_from(@configuration.root_path, 
package_pathspec: package_glob)
  this_package = package_set.package_from_path(relative_path(filepath))
  constant_package = package_set.package_from_path(constant_context.location)

  unless constant_package == this_package
    errors << "'#{constant}' is declared as private in the '#{this_package}' package but appears to be "\
        "defined\nin the '#{constant_package}' package. Packwerk resolved it to #{constant_context.location}."
  end
end
```

Reply:
that would be called in the loop iterating over all packages, and re-initialize the package set for each iteration.

Reply:
I did a refactor. Had to make quite a few modifications to make it work.